app/src/test/java/com/github/bsideup/liiklus/GRPCAuthTest.java

```diff
@@ -0,0 +1,195 @@
+package com.github.bsideup.liiklus;
+
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.algorithms.Algorithm;
+import com.auth0.jwt.interfaces.RSAKeyProvider;
+import com.avast.grpc.jwt.client.JwtCallCredentials;
+import com.github.bsideup.liiklus.protocol.PublishRequest;
+import com.google.protobuf.ByteString;
+import io.grpc.CallOptions;
+import io.grpc.Channel;
+import io.grpc.ClientCall;
+import io.grpc.ClientInterceptor;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.MethodDescriptor;
+import io.grpc.Server;
+import io.grpc.StatusRuntimeException;
+import lombok.SneakyThrows;
+import org.junit.Test;
+import org.pf4j.PluginManager;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ConfigurableApplicationContext;
+import reactor.core.publisher.Hooks;
+
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.security.KeyFactory;
+import java.security.interfaces.RSAPrivateKey;
+import java.security.interfaces.RSAPublicKey;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.util.Base64;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class GRPCAuthTest {
+
+    static {
+        System.setProperty("plugins.dir", "../plugins");
+        System.setProperty("plugins.pathMatcher", "*/build/libs/*.jar");
+
+        Hooks.onOperatorDebug();
+    }
+
+    static ConfigurableApplicationContext startLiiklus(List<String> args) {
```

Comment:
I wonder if `ApplicationRunner` from `tck` can be used here

Reply:
you mean depend on it? Or copy? This is basically limited copy